### PR TITLE
feat(headers): add AcceptLanguage header

### PR DESCRIPTION
**Pull request by hugoduncan**
Adds header support for Accept-Language, using a struct for the primary and
sub components of the language.

NOTE: I have been unable to get equality comparison working

### CODE REVIEW

File: src/header/common/accept_language.rs [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+pub struct Language{
+    primary: String,
+    sub: Option<String>
+}
+
+impl FromStr for Language {
+    type Err = ();
+    fn from_str(s: &str) -> Result<Language, ()> {
+        let mut i=s.split_str("-");
+        let p=i.next();
```
Comment (seanmonstar) [inline src/header/common/accept_language.rs]: Nit: space before and after = signs

File: src/header/common/accept_language.rs [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+        let s=i.next();
+        match (p,s) {
```
Comment (seanmonstar) [inline src/header/common/accept_language.rs]: Nit: space after commas.

File: src/header/common/accept_language.rs [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+impl_list_header!(AcceptLanguage,
+                  "Accept-Language",
+                  Vec<QualityItem<Language>>);
+
+#[test]
+fn test_parse_header() {
+    let a: AcceptLanguage = header::Header::parse_header(
+        [b"en-us;q=1.0, en;q=0.9, fr".to_vec()].as_slice()).unwrap();
+    let b = AcceptLanguage(vec![
+        QualityItem{item: Language{primary: "en".to_string(),
```
Comment (seanmonstar) [inline src/header/common/accept_language.rs]: Nit: space before and after { braces.

**Comment by seanmonstar** [comment]
What's the issue with equality? The floats?

File: src/header/common/accept_language.rs [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+                                   sub: Some("us".to_string())},
+                    quality: 1f32},
+        QualityItem{item: Language{primary: "en".to_string(), sub: None},
+                    quality: 0.9f32},
+        QualityItem{item: Language{primary: "fr".to_string(), sub: None},
+                    quality: 1f32},
+    ]);
+    assert_eq!(format!("{}", a), format!("{}", b));
+    // assert_eq!(a, b);
```
Comment (hugoduncan) [inline src/header/common/accept_language.rs]: @seanmonstar Uncommenting this line fails, and I'm not sure why.

File: src/header/common/accept_language.rs (hunk 3 above)
Comment (hugoduncan) [inline src/header/common/accept_language.rs]: Just to be sure, it should be like this?

```
QualityItem { item … quality: 0.9f32 },
```

Comment (seanmonstar) [inline src/header/common/accept_language.rs]: Correct.

**Comment by pyfisch** [comment]
Merge this now. I will update language-tags at a later date.

**Comment by seanmonstar** [comment]
It seems there's an issue with Eq in this PR...

**Comment by GitCop** [comment]
Thanks for contributing! Unfortunately, I'm here to tell you there were the following style issues with your Pull Request:

 * Commit: 97a9f5d207280c950797e1d8d92437234d1a2151
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

 * Commit: e62cdfd7af7c3cd47247ef69b93f6da03357c058
  * Commits must be in the following format: **%{type}(%{scope}): %{description}**

Guidelines are available at https://github.com/hyperium/hyper/blob/master/CONTRIBUTING.md

---
This message was auto-generated by https://gitcop.com

**Comment by hugoduncan** [comment]
The equality issue is in `QualityItem` for `quality` values that can't be exactly represented by `f32`.

I added a `bench`, squashed, and rebased.

**Pull request merged**